### PR TITLE
Replace fpm

### PR DESCRIPTION
**Pull request by GeorgeJahad**
Replaced fpm with nfpm, because fpm had bugs: https://jira.rax.io/browse/CMC-1736
To completely emulate fpm behaviour, I had to add some postinstall/preremove scripts to the git repo. I copied them verbatim from the fpm output.

FPM was also modifying the permissions of the output files; since nfpm doesn't, and I wanted to match the FPM behaviour as closely as possible I added some chmod's to the makefile.

Installed and ran some simple tests to confirmed the poller still works, and there are no errors returned by dpkg.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,8 +37,74 @@ DEB_CONFIG_FILES := ${APP_CFG} ${LOGROTATE_CFG}
 
 PKG_VERSION := ${GIT_TAG}-${TAG_DISTANCE}
 PKG_BASE := ${APP_NAME}_${PKG_VERSION}_${ARCH}
-FPM_IDENTIFIERS := -n ${APP_NAME} --license "${LICENSE}" --vendor "${VENDOR}" \
-                   	  -v ${GIT_TAG} --iteration ${TAG_DISTANCE}
+
+
+define YAML_HDR
```
Comment (itzg) [inline Makefile]: Nice use of define's!

File: Makefile [hunk 2]
```diff
@@ -148,32 +217,26 @@ package-debs: ${BUILD_DIR}/${PKG_BASE}.deb \
...
-	  ${FPM_IDENTIFIERS} \
-	  --deb-default ${DEB_BUILD_DIR}/${UPSTART_DEFAULT} \
-	  --deb-upstart ${DEB_BUILD_DIR}/${UPSTART_CONF} \
-	  --deb-systemd ${DEB_BUILD_DIR}/${SYSTEMD_CONF} \
-	  --no-deb-systemd-restart-after-upgrade \
-	  -C ${DEB_BUILD_DIR} ${PKGDIR_BIN}/${EXE} ${DEB_CONFIG_FILES}
-
-${BUILD_DIR}/${PKG_BASE}_systemd.deb : $(addprefix ${DEB_BUILD_DIR}/,${PKGDIR_BIN}/${EXE} ${DEB_CONFIG_FILES} ${SYSTEMD_CONF})
+	echo "$$DEB_YAML" > ${BUILD_DIR}/deb.yaml
+	${NFPM} -f ${BUILD_DIR}/deb.yaml pkg -t $@
```
Comment (itzg) [inline Makefile]: The command line is certainly nicer with nfpm.

File: pkg/debian/postinst [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
```
Comment (itzg) [inline pkg/debian/postinst]: Does nfpm not generate these or were they not suitably complete?

Comment (GeorgeJahad) [reply pkg/debian/postinst]: It doesn't generate them.

**Pull request merged**